src/core/idb-cache.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * A promise based interface to a key-value based indexedDB cache
+ */
+
+const databases = new Map();
+
+export default class IDBCache {
+  /**
+   * @constructor
+   * @param {string} name             name of database
+   * @param {array} stores            array of names of stores
+   * @param {object} Options
+   *    @param {integer} version      database version
+   *    @param {string} defaultStore  name of default store
+   */
+  constructor(
+    name,
+    stores,
+    { version, defaultStore } = { version: 1, defaultStore: null }
+  ) {
+    this.name = name;
+    if (!Array.isArray(stores) || stores.length < 1) {
+      throw new TypeError("`stores` must be an array of length atleast 1");
+    }
+    this.stores = [...stores];
+    this.version = version;
+    this.defaultStore = defaultStore || this.stores[0];
+    if (!databases.has(name)) {
+      const dbPromise = new Promise((resolve, reject) => {
+        const request = window.indexedDB.open(this.name, this.version);
+        request.onerror = () => {
+          reject(new DOMException(request.error.message, request.error.name));
+        };
+        request.onupgradeneeded = () => {
+          this.stores.forEach(storeName =>
+            request.result.createObjectStore(storeName));
+        };
+        request.onsuccess = () => resolve(request.result);
+      });
+      databases.set(this.name, dbPromise);
+    }
+  }
+
+  get ready() {
+    return new Promise(async (resolve) => {
+      const db = await databases.get(this.name);
```

Comment:
Generally, a getter shouldn’t return a new instance. This is why we should set it via `this.ready = ... `.

Reply:
I'm not clearly understanding this one thing that you're saying. Can you edit the code for this?
I've added a separate PR for IDBCache here: https://github.com/w3c/respec/pull/1586

Reply:
Will do!